### PR TITLE
Make pressure clipping in `BoundaryModelDummyParticles` optional

### PR DESCRIPTION
**Pull request by efaulhaber**
Pressure clipping is not a correct formulation, it is just a workaround to avoid sticking artifacts.
Away from free surfaces it leads to artificially increased boundary pressures that cause larger gaps between fluid and boundary in areas of low pressure, against which the particle shifting technique is fighting.

This PR adds an option to disable clipping, which should be enabled for closed systems.

### CODE REVIEW

**Comment by svchb** [comment]
What is supposed to happen when clipping in the state equation is activated and deactivated here?

**Comment by efaulhaber** [comment]
When clipping in the state equation is activated, then pressure is clipped everywhere and it cannot become negative anyway, except perhaps with Adami when the boundary is accelerated away from the fluid.

**Comment by svchb** [comment]
But shouldn't this be an error than or warning?

**Comment by efaulhaber** [comment]
> But shouldn't this be an error than or warning?

Why? Everything is working as expected.

**Comment by svchb** [comment]
Because the inconsistency might be confusing.

**Comment by efaulhaber** [comment]
I don't see an inconsistency. Enable pressure clipping for the boundary model to clip boundary pressure. Enable pressure clipping of the state equation to enable global pressure clipping.

**Comment by svchb** [comment]
Ok than just adjust the docstring also for state equation pressure clipping.

File: src/schemes/boundary/wall_boundary/dummy_particles.jl [hunk 1]
```diff
@@ -22,6 +23,15 @@ Boundary model for [`WallBoundarySystem`](@ref).
 - `correction`:                 Correction method of the adjacent fluid system (see [Corrections](@ref corrections)).
 - `viscosity`:                  Slip (default) or no-slip condition. See description below for further
                                 information.
+- `clip_negative_pressure=true`: Clip negative boundary pressures to avoid sticking
```
Comment (LasNikas) [inline src/schemes/boundary/wall_boundary/dummy_particles.jl]: As discussed on Slack, I would set the default to `false` if it isn't explicitly mentioned anywhere in the literature.

**Review by LasNikas — CHANGES_REQUESTED** [review]

File: src/schemes/boundary/wall_boundary/dummy_particles.jl (hunk 1 above)
Comment (efaulhaber) [reply src/schemes/boundary/wall_boundary/dummy_particles.jl]: Done.

File: src/schemes/boundary/wall_boundary/dummy_particles.jl [hunk 2]
```diff
@@ -22,6 +23,15 @@ Boundary model for [`WallBoundarySystem`](@ref).
...
                                 information.
+- `clip_negative_pressure=false`: Clip negative boundary pressures to avoid sticking
+                                artifacts from attractive fluid-boundary forces at free
+                                surfaces. Note that this is not a correct formulation
+                                at interior boundaries (away from free surfaces).
+                                Disable this option when simulating closed systems without
+                                free surfaces to avoid artificially increased boundary
+                                pressures that cause larger gaps between fluid and boundary
+                                in areas of low pressure, against which the particle
+                                shifting technique is fighting.
```
Comment (svchb) [inline src/schemes/boundary/wall_boundary/dummy_particles.jl:34]:
```suggestion
- `clip_negative_pressure=false`: Clip negative boundary pressures to avoid sticking
                                artifacts from attractive fluid-boundary forces at free
                                surfaces. Note that this is not a correct formulation
                                at interior boundaries (away from free surfaces).

                                This option only controls clipping performed inside
                                `BoundaryModelDummyParticles` itself. If the selected
                                `state_equation` already clips negative pressures,
                                then boundary pressures computed directly from that
                                state equation can still be non-negative regardless
                                of this setting.

                                Disable this option when simulating closed systems without
                                free surfaces to avoid artificially increased boundary
                                pressures that cause larger gaps between fluid and boundary
                                in areas of low pressure, against which the particle
                                shifting technique is fighting.
```

**Review by svchb — CHANGES_REQUESTED** [review]
- Add a similar note to state equation
- add a test for the mixed stateequation/boundary clipping

**Review by LasNikas — APPROVED** [review]

**Review by svchb — APPROVED** [review]
